buffer/src/main/java/io/netty/buffer/PoolArena.java

```diff
@@ -328,25 +641,52 @@ public synchronized String toString() {
                 }
             }
         }
-        buf.append(StringUtil.NEWLINE);
+    }
 
-        return buf.toString();
+    @Override
+    protected final void finalize() throws Throwable {
+        try {
+            super.finalize();
+        } finally {
+            destroyPoolSubPages(smallSubpagePools);
+            destroyPoolSubPages(tinySubpagePools);
+            destroyPoolChunkLists(qInit, q000, q025, q050, q075, q100);
+        }
+    }
+
+    private static void destroyPoolSubPages(PoolSubpage<?>[] pages) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Consider using varargs for methods or constructors which take an array the last parameter. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUseVarargs)